scipy/sparse/linalg/_isolve/symmlq.py

```diff
@@ -0,0 +1,239 @@
+import numpy as np
+from .utils import make_system
+
+
+__all__ = ['symmlq']
+
+
+def symmlq(A, b, x0=None, tol=1e-5, maxiter=None, M=None, callback=None,
+           atol=None, haptol=1e-18, show=False):
+    """
+    Use Symmetric LQ iteration to solve ``Ax = b``.
+
+    Parameters
+    ----------
+    A : {sparse matrix, ndarray, LinearOperator}
+        The real or complex N-by-N matrix of the linear system.
+        Alternatively, `A` can be a linear operator which can
+        produce ``Ax`` using, e.g.,
+        `scipy.sparse.linalg.LinearOperator`.
+    b : {ndarray}
+        Right hand side of the linear system. Has shape (N,) or (N, 1).
+    x0 : {ndarray}
+        Starting guess for the solution.
+    tol, atol : float, optional
+        Tolerances for convergence, ``norm(b - Ax) < max(tol*norm(b), atol)``.
+        The default for `tol` is 1.0e-5.
+        The default for `atol` is ``tol * norm(b)``.
```

Comment:
Choice for atol is strange why not 0. if you want to turn it off by default?

Reply:
The atol default is either None or 0. What happens to the collective tolerance is independent from this number. You need to state the value of the keyword not its effect.